### DISCUSSION TITLE
"Enter new value" box to reprice items should spawn unfilled

### DISCUSSION THREAD

**Discussion by harvito** [OpenRCT2/OpenRCT2#22126] [Feature requests]
### Operating System

Win11 64

### OpenRCT2 build

OpenRCT2, v0.4.11 (18d2b5e) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

The bug also exists in the original game (RCT2/RCTC)

### Describe the issue

"Enter new value" and all rename boxes come prefilled. Their contents are almost always completely removed to make way for a new value. The box should instead spawn unfilled, making it easier to immediately type the new value. The player loses no information, as the object's current name is visible in its window title, and of course the cancel button would still work.
![image](https://github.com/OpenRCT2/OpenRCT2/assets/14914822/838f153f-4dbd-4881-9563-b863367868da)
![image](https://github.com/OpenRCT2/OpenRCT2/assets/14914822/d06ae2ca-79cc-4250-ba05-9a887487b32b)
Boxes that should spawn unfilled: reprice ride, shop item, on ride photo, park entrace, and rename guest, staff, ride, park, signs
Boxes that should spawn filled: Save track design, save game new file
There are other boxes that should stay the same too, like Enter name for scenario chart, and I'm not sure about:
- Save landscape currently prefills "My new scenario"

### Steps to reproduce

Reprice or rename anything

**Comment by Lastorder-DC** [comment]
I think adding clear button next to input is more reasonable than just removing default value as emptying default value makes hard to modify ride's name.

**Reply by Gymnasiast** [reply]
That sounds like a much better solution to me as well.

**Comment by harvito** [comment]
It's a good point, even just for those rare times (in my experience) one needs to ex fix a typo they previously made. It would be possible to implement if is default name then spawn unfilled else spawn filled for names.

**Comment by Gymnasiast** [comment]
This is not a bug, so it doesn’t belong under issues. Moved it to Discussions.

**Comment by MASHzj** [comment]
I disagree with renaming things. For example I always remove the "1" from my rides where I only have one of that type. So I wouldn't want to type those all over. And if want a plain new name, I use ctrl+backspace. But for numbers it would indeed be better.

**Reply by harvito** [reply]
Thanks for the feedback, and for the tip about ctrl + backspace. Do you know a key combo for deleting the whole name? ctrl+backspace does just one word at a time. If not, ctrl+shift+backspace could be a nice enhancement.

**Reply by harvito** [reply]
I made a new discussion about ctrl+shift+backspace: https://github.com/OpenRCT2/OpenRCT2/discussions/22156

**Comment by harvito** [comment]
It seems removing the name prefill is not desirable, I'm happy to resolve that part as won't implement. I am still curious about repricing boxes, however, and will rename this discussion to focus on this.
I'll speak to how I reprice items:
- small change: use the mouse wheel or +- buttons inside the price box
- large change, as when pricing a new roller coaster: open Enter new value box, clear it, enter price then enter
- max price, as in umbrellas: open Enter new value box, clear it, enter 99 then enter
So basically every time I use the Enter new value box, I first clear it.

Is the prefill for this box useful for anyone?
